modal/_runtime/gpu_memory_snapshot.py

```diff
@@ -125,75 +161,143 @@ def _get_cuda_pids(self) -> list[CudaCheckpointProcess]:
                 "OS does not have /proc path rendering it incompatible with GPU memory snapshots."
             )
 
-        for entry in proc_dir.iterdir():
-            if not entry.name.isdigit():
-                continue
-
-            pid = int(entry.name)
-            try:
-                # Call cuda-checkpoint to check if this PID has a CUDA session
-                result = subprocess.run(
-                    [CUDA_CHECKPOINT_PATH, "--get-state", "--pid", str(pid)],
-                    capture_output=True,
-                    text=True,
-                    timeout=10,
-                )
-
-                # If the command succeeds (return code 0), this PID has a CUDA session
-                if result.returncode == 0:
-                    state_str = result.stdout.strip().lower()
-                    state = CudaCheckpointState(state_str)
-
-                    cuda_checkpoint_process = CudaCheckpointProcess(pid=pid, state=state)
-                    cuda_pids.append(cuda_checkpoint_process)
+        # Get all numeric directories (PIDs) from /proc
+        pid_dirs = [entry for entry in proc_dir.iterdir() if entry.name.isdigit()]
```

Comment:
@luiscape Some PIDs can be CPU-only processes without GPU state. In the CUDA Plugin for CRIU we use the `--get-restore-tid` option to retrieve the cuda TIDs: https://github.com/checkpoint-restore/criu/blob/57251d8/plugins/cuda/cuda_plugin.c#L211